src/components/input/InputGroup.tsx

```diff
@@ -12,7 +12,9 @@ export const inputGroupStyles = classnames(
   // Restore border-radius on the leftmost and rightmost components in the group
   'input-group:first:rounded-l input-group:last:rounded-r',
   // "Collapse" borders between input components
-  'input-group:border-l-0 input-group:first:border-l',
+  'input-group:ml-[-1px] input-group:first:ml-0',
```

Comment:
To make sure the whole select borders change color even when used in an input group, we collapse the items  via negative margin instead of by removing borders.

Reply:
Incomplete border without this change:

<img width="664" alt="Cropped border" src="https://github.com/user-attachments/assets/c938263b-7f21-4709-9e2a-14eede77298b">

The code assumes that the border width is 1px, if we ever change that we'll have to adapt both places.